### PR TITLE
Merge develop into main 

### PR DESCRIPTION
**Pull request by davidcui-amzn**
### Description
Merge `develop` into `main` and delete `develop` to use `main` as development branch moving forward. Will cut 1.0 and 1.x branches as well as per #135 
 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]

**Comment by chloe-zh** [comment]
Could you help deal with #128 before merge dev to main, I mistakenly merged the PR to main days before

**Comment by joshuali925** [comment]
> Could you help deal with #128 before merge dev to main, I mistakenly merged the PR to main days before

I checked #128 as well, but this PR merges develop to main, I don't think it is affected?

**Comment by chloe-zh** [comment]
> > Could you help deal with #128 before merge dev to main, I mistakenly merged the PR to main days before
> 
> I checked #128 as well, but this PR merges develop to main, I don't think it is affected?

My bad, I thought we would still use the dev branch as our default branch, so it should be always consistent or ahead of main. Yes you are right, it does not matter if we are removing dev branch.

**Review by chloe-zh — APPROVED** [review]
LGTM once the link checker is fixed. Thanks!

File: sql-odbc/docs/dev/sign_installers.md
Comment (joshuali925) [inline sql-odbc/docs/dev/sign_installers.md:9]: probably better to update the instruction? instead of removing it
https://docs.microsoft.com/en-us/powershell/module/pki/import-certificate?view=windowsserver2019-ps

Comment (davidcui-amzn) [reply sql-odbc/docs/dev/sign_installers.md:9]: I could not find an updated link for this, think this should be safe to remove since it is all internal-facing, please confirm @penghuo

Comment (joshuali925) [reply sql-odbc/docs/dev/sign_installers.md:9]: I posted the updated link in my comment? I believe it's just the doc for `Import-Certificate` powershell command, url says `sindowsserver2019-ps` but this also applies to win10-ps

Comment (davidcui-amzn) [reply sql-odbc/docs/dev/sign_installers.md:9]: Whoops, i thought you had just copy-pasted the original link- sure, can replace

Comment (penghuo) [reply sql-odbc/docs/dev/sign_installers.md]: We are good to remove the instruction, customer should not sign it.

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Pull request merged**
